File: auth-api/src/main/java/com/cartobucket/auth/api/server/exceptions/IllegalArgumentExceptionMapper.java
package com.cartobucket.auth.api.server.exceptions;

import jakarta.ws.rs.core.Response;
import jakarta.ws.rs.ext.ExceptionMapper;
import jakarta.ws.rs.ext.Provider;

@Provider
public class IllegalArgumentExceptionMapper implements ExceptionMapper<IllegalArgumentException> {
    @Override
    public Response toResponse(IllegalArgumentException exception) {
        return Response.status(400).entity(exception.getMessage()).build();
    }
}

File: auth-api/src/main/kotlin/com/cartobucket/auth/api/server/exceptions/IllegalArgumentExceptionMapper.kt
package com.cartobucket.auth.api.server.exceptions

import jakarta.ws.rs.core.Response
import jakarta.ws.rs.ext.ExceptionMapper
import jakarta.ws.rs.ext.Provider

@Provider
class IllegalArgumentExceptionMapper : ExceptionMapper<IllegalArgumentException> {
    override fun toResponse(exception: IllegalArgumentException): Response {
        return Response.status(400).entity(exception.message).build()
    }
}
